Add supporting custom install prefix for native on Linux

ava57r opened the pull request:
Hello 
I tried to build this crate with builded from source leptonica

I used script like this

```
wget http://www.leptonica.org/source/leptonica-1.79.0.tar.gz
tar -zxvf leptonica-1.79.0.tar.gz
cd leptonica-1.79.0
./configure --prefix=${HOME}/local/leptonica-1.79.0
make
make install
```
After that tried to build the crate.
It failed

```
--- stderr
  wrapper.h:1:10: fatal error: 'leptonica/allheaders.h' file not found
  wrapper.h:1:10: fatal error: 'leptonica/allheaders.h' file not found, err: true
  thread 'main' panicked at 'Unable to generate bindings: ()', build.rs:60:10
```
I added `pkg_config` for Linux
And setted before running `cargo build`

```shell
export PKG_CONFIG_PATH=${HOME}/local/leptonica-1.80.0/lib/pkgconfig
```

It allow to change install prefix for native leptonica.
If  `PKG_CONFIG_PATH` not set It works with system lib.
I checked on Ubuntu 20.04 and lib `libleptonica-dev`

ava57r committed:
Add supporting custom install prefix for native on Linux

ava57r committed:
Fix formatting

ccouzens commented on Cargo.toml:
```diff
@@ -18,4 +18,7 @@ bindgen = "0.56"
 vcpkg = "0.2.8"
 
 [target.'cfg(target_os="macos")'.build-dependencies]
+pkg-config = "0.3.19"
+
+[target.'cfg(target_os="linux")'.build-dependencies]
```
I think this might work- I'll try it later
```suggestion
[target.'cfg(any(target_os="macos", target_os="linux"))'.build-dependencies]
pkg-config = "0.3.19"
```

ccouzens commented on build.rs:
```diff
@@ -39,7 +39,21 @@ fn find_leptonica_system_lib() -> Option<String> {
 }
 
-#[cfg(all(not(windows), not(target_os = "macos")))]
+#[cfg(target_os = "linux")]
```
Is this the same as the MacOS pkg-config search?

Can we delete one of them and use this `cfg` line?

```suggestion
#[cfg(any(target_os = "linux", target_os = "linux"))]
```

ccouzens commented:
Thank you for this :)

ava57r committed:
Fix PR suggestions

ava57r committed:
revert pkg-config version

conflicting with published version leptonica-sys
when use [patch] section on Cargo.toml
